fix bug downsampleing wsi images to height x width which are smaller than source frame

Philbrick opened the pull request:
fix bug downsampleing wsi images to height x width which are smaller than frame (height x widht)

Philbrick committed:
fix downsampleing wsi images to heightxwidth which are smaller than the specified frame size.

danielbeaudreau commented:
/gcbrun

danielbeaudreau commented:
some linter errors:

```
././src/geometryUtils.cpp:25:  Tab found; better to use spaces  [whitespace/tab] [1]
././src/geometryUtils.h:37:  Tab found; better to use spaces  [whitespace/tab] [1]
././src/wsiToDcm.cpp:214:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
././src/wsiToDcm.cpp:217:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
././src/wsiToDcm.cpp:227:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
././src/wsiToDcm.cpp:241:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
././src/wsiToDcm.cpp:262:  Lines should be <= 80 characters long  [whitespace/line_length] [2]

danielbeaudreau commented:
The end to end script segfaults on this change:

```
adduser: Warning: The home directory `/var/lib/dcmtk/db' does not belong to the user you are currently creating.
Processing triggers for libc-bin (2.31-0ubuntu9.2) ...
./endToEnd/endToEndTest.sh: line 36: 12455 Segmentation fault      (core dumped) ./build/wsi2dcm $fileName ./endToEnd/ --seriesDescription test1 --levels 9 --startOn 8 --compression jpeg2000
E: dcm2json: error (End of stream) reading file: ./endToEnd/level-8-frames-0-1.dcm
diff: ./endToEnd/test1GeneratedTags.json: No such file or directory
[2021-05-27 14:41:43.215345] [0x00007fb12789ea40] [warning] studyId is going to be generated
[2021-05-27 14:41:43.215368] [0x00007fb12789ea40] [warning] seriesId is going to be generated
[2021-05-27 14:41:43.215370] [0x00007fb12789ea40] [warning] threads parameter is less than 1, consuming all avalible threads
[2021-05-27 14:41:43.215372] [0x00007fb12789ea40] [warning] batch parameter is not set, batch is unlimited
[2021-05-27 14:41:43.225834] [0x00007fb12789ea40] [info]    dicomization is started
[2021-05-27 14:41:43.798594] [0x00007fb12789ea40] [info]    dicomization is done
./endToEnd/endToEndTest.sh: line 46: 12469 Floating point exception(core dumped) ./build/wsi2dcm $fileName ./endToEnd/ --seriesDescription test3 --levels 15 --startOn 14 --tileHeight 100 --tileWidth 100 --compression raw
F: End of stream: reading file: ./endToEnd/level-14-frames-0-1.dcm
diff: ./endToEnd/test3GeneratedImage.ppm: No such file or directory
./endToEnd/endToEndTest.sh: line 51: 12474 Floating point exception(core dumped) ./build/wsi2dcm $fileName ./endToEnd/ --seriesDescription test4 --levels 15 --startOn 14 --tileHeight 100 --tileWidth 100 --compression raw --dropFirstRowAndColumn
F: End of stream: reading file: ./endToEnd/level-14-frames-0-1.dcm
diff: ./endToEnd/test4GeneratedImage.ppm: No such file or directory
```

Please fix.

Philbrick committed:
Add workaround for issue with JPEG2000 codec

JPEG 2000 codec seg faults when compressing very small images. Fix provides a workaround making sure that images do not get too small.

danielbeaudreau commented:
/gcbrun

danielbeaudreau commented:
linter failed:

```
././src/geometryUtils.cpp:51:  Add #include <algorithm> for max  [build/include_what_you_use] [4]
././src/wsiToDcm.cpp:214:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
././src/wsiToDcm.cpp:217:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
././src/wsiToDcm.cpp:227:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
././src/wsiToDcm.cpp:241:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
././src/wsiToDcm.cpp:262:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
```

danielbeaudreau commented:
Test presubmit fails with the following error:

```
Processing triggers for libc-bin (2.31-0ubuntu9.2) ...
[2021-06-04 15:56:48.741345] [0x00007f442b4eca40] [warning] studyId is going to be generated
[2021-06-04 15:56:48.741369] [0x00007f442b4eca40] [warning] seriesId is going to be generated
[2021-06-04 15:56:48.741384] [0x00007f442b4eca40] [warning] threads parameter is less than 1, consuming all avalible threads
[2021-06-04 15:56:48.741386] [0x00007f442b4eca40] [warning] batch parameter is not set, batch is unlimited
[2021-06-04 15:56:48.753386] [0x00007f442b4eca40] [info]    dicomization is started
[2021-06-04 15:56:49.161614] [0x00007f442b4eca40] [info]    dicomization is done
> 500 > 500
ERROR
ERROR: build step 0 "ubuntu" failed: step exited with non-zero status: 1
```
Can you confirm that the end to end test passes locally?

Philbrick commented:
Yes, end-to-end tests ran fine.
for some reason I do see a get a c++: fatal error: Killed signal terminated program cc1plus.  But again, tests appear fine.



replace jpeg-9c/aclocal.m4? [y]es, [n]o, [A]ll, [N]one, [r]ename: N
/wsi-to-dicom-converter/build
-- Info: DCMTK will compile with built-in (compiled-in) dictionary
-- Info: DCMTK will not try to load external dictionary from default path on startup
-- Info: DCMTK's builtin private dictionary support will be disabled
-- Info: Thread support will be enabled
-- Info: Wide char file I/O functions will be disabled
-- Info: Wide char main function for command line tools will be disabled
-- Info: Building DCMTK without character set conversion support
-- Info: Building DCMTK with explicit large file support (LFS64)
-- Info: C++11 features enabled
-- Info: STL vector support disabled
-- Info: STL algorithm support disabled
-- Info: STL limits support disabled
-- Info: STL list support disabled
-- Info: STL map support disabled
-- Info: STL memory support disabled
-- Info: STL stack support disabled
-- Info: STL string support disabled
-- Info: STL type_traits support disabled
-- Info: STL tuple support disabled
-- Info: STL system_error support disabled
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... no
checking for mawk... mawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make supports nested variables... (cached) yes
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking for style of include used by make... GNU
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking for gcc option to accept ISO C99... none needed
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... no
checking for _LARGEFILE_SOURCE value needed for large files... no
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for uintptr_t... yes
checking how to print strings... printf
checking for a sed that does not truncate output... /bin/sed
checking for fgrep... /bin/grep -F
checking for ld used by gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... yes
checking how to convert x86_64-unknown-linux-gnu file names to x86_64-unknown-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-unknown-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for mt... no
checking if : is a manifest tool... no
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... yes
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
checking for windres... no
checking version string... 3.4.1
checking for library containing floor... -lm
checking for library containing jpeg_CreateDecompress... -ljpeg
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for ZLIB... yes
checking for OPENJPEG2... yes
checking for LIBTIFF... yes
checking for GLIB2... yes
checking for CAIRO... yes
checking for LIBPNG... yes
checking for GDKPIXBUF... yes
checking for LIBXML2... yes
checking for SQLITE3... yes
checking for VALGRIND... yes
checking for simple visibility declarations... yes
checking fopen() close-on-exec flag... e
checking for fcntl... yes
checking whether to cross-test from Cygwin... no
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating openslide.pc
config.status: creating src/openslide-dll.manifest
config.status: creating src/openslide-dll.rc
config.status: creating tools/openslide-quickhash1sum.1
config.status: creating tools/openslide-show-properties.1
config.status: creating tools/openslide-write-png.1
config.status: creating config.h
config.status: config.h is unchanged
config.status: executing depfiles commands
config.status: executing libtool commands
-- Configuring done
-- Generating done
-- Build files have been written to: /wsi-to-dicom-converter/build
Scanning dependencies of target wsi2dcm
[  0%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/cmdlnarg.cc.o
[  0%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dccodec.cc.o
[  1%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcbytstr.cc.o
[  1%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcchrstr.cc.o
[  2%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcdatset.cc.o
[  2%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcdatutl.cc.o
[  3%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcdicdir.cc.o
[  3%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcddirif.cc.o
[  3%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcdicent.cc.o
[  3%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcdict.cc.o
[  3%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcdictbi.cc.o
[  4%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcdirrec.cc.o
[  4%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcelem.cc.o
[  4%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcerror.cc.o
[  5%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcfilefo.cc.o
In function ‘int splitFields(const char*, char**, int, char)’,
    inlined from ‘OFBool DcmDataDictionary::loadDictionary(const char*, OFBool)’ at /wsi-to-dicom-converter/build/dcmtk-3.6.2/dcmdata/libsrc/dcdict.cc:476:36:
/wsi-to-dicom-converter/build/dcmtk-3.6.2/dcmdata/libsrc/dcdict.cc:246:16: warning: ‘char* strncpy(char*, const char*, size_t)’ specified bound depends on the length of the source argument [-Wstringop-overflow=]
         strncpy(fields[foundFields], line, len);
         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/wsi-to-dicom-converter/build/dcmtk-3.6.2/dcmdata/libsrc/dcdict.cc: In member function ‘OFBool DcmDataDictionary::loadDictionary(const char*, OFBool)’:
/wsi-to-dicom-converter/build/dcmtk-3.6.2/dcmdata/libsrc/dcdict.cc:241:25: note: length computed here
             len = strlen(line);
                   ~~~~~~^~~~~~
[  5%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcfilter.cc.o
[  5%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dchashdi.cc.o
In function ‘int splitFields(const char*, char**, int, char)’,
    inlined from ‘OFBool DcmDataDictionary::loadExternalDictionaries()’ at /wsi-to-dicom-converter/build/dcmtk-3.6.2/dcmdata/libsrc/dcdict.cc:619:37:
/wsi-to-dicom-converter/build/dcmtk-3.6.2/dcmdata/libsrc/dcdict.cc:246:16: warning: ‘char* strncpy(char*, const char*, size_t)’ specified bound depends on the length of the source argument [-Wstringop-overflow=]
         strncpy(fields[foundFields], line, len);
         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/wsi-to-dicom-converter/build/dcmtk-3.6.2/dcmdata/libsrc/dcdict.cc: In member function ‘OFBool DcmDataDictionary::loadExternalDictionaries()’:
/wsi-to-dicom-converter/build/dcmtk-3.6.2/dcmdata/libsrc/dcdict.cc:241:25: note: length computed here
             len = strlen(line);
                   ~~~~~~^~~~~~
[  6%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcistrma.cc.o
[  6%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcistrmb.cc.o
[  6%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcistrmf.cc.o
[  6%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcistrmz.cc.o
[  7%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcitem.cc.o
[  7%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcjson.cc.o
[  7%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dclist.cc.o
[  8%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcmatch.cc.o
[  8%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcmetinf.cc.o
[  8%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcobject.cc.o
[  9%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcostrma.cc.o
[  9%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcostrmb.cc.o
[  9%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcostrmf.cc.o
[  9%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcostrmz.cc.o
[ 10%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcpath.cc.o
[ 10%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcpcache.cc.o
[ 10%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcpixel.cc.o
[ 11%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcpixseq.cc.o
[ 11%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcpxitem.cc.o
[ 11%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcrleccd.cc.o
[ 11%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcrlecce.cc.o
[ 12%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcrlecp.cc.o
[ 12%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcrledrg.cc.o
[ 12%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcrleerg.cc.o
[ 13%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcrlerp.cc.o
[ 13%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcsequen.cc.o
[ 13%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcspchrs.cc.o
[ 14%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcstack.cc.o
[ 14%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcswap.cc.o
[ 14%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dctag.cc.o
[ 14%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dctagkey.cc.o
[ 15%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dctypes.cc.o
[ 15%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcuid.cc.o
[ 15%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvr.cc.o
[ 16%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvrae.cc.o
[ 16%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvras.cc.o
[ 16%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvrat.cc.o
[ 17%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvrcs.cc.o
[ 17%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvrda.cc.o
[ 17%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvrds.cc.o
[ 17%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvrdt.cc.o
[ 18%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvrfd.cc.o
[ 18%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvrfl.cc.o
[ 18%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvris.cc.o
[ 19%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvrlo.cc.o
[ 19%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvrlt.cc.o
[ 19%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvrobow.cc.o
[ 19%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvrof.cc.o
[ 20%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvrod.cc.o
[ 20%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvrol.cc.o
[ 20%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvrpn.cc.o
[ 21%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvrpobw.cc.o
[ 21%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvrsh.cc.o
[ 21%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvrsl.cc.o
[ 22%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvrss.cc.o
[ 22%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvrst.cc.o
[ 22%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvrtm.cc.o
[ 23%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvruc.cc.o
[ 23%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvrui.cc.o
[ 23%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvrul.cc.o
[ 23%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvrulup.cc.o
[ 24%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvrur.cc.o
[ 24%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvrus.cc.o
[ 24%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcvrut.cc.o
[ 25%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcwcache.cc.o
[ 25%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/dcxfer.cc.o
[ 25%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/vrscan.cc.o
[ 25%] Building C object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libsrc/vrscanl.c.o
[ 26%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libi2d/i2d.cc.o
[ 26%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libi2d/i2dbmps.cc.o
[ 26%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libi2d/i2djpgs.cc.o
[ 27%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libi2d/i2dplnsc.cc.o
[ 27%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libi2d/i2dplsc.cc.o
[ 28%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/ofstd/libsrc/ofchrenc.cc.o
[ 28%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/dcmdata/libi2d/i2dplvlp.cc.o
[ 28%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/ofstd/libsrc/ofcmdln.cc.o
[ 28%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/ofstd/libsrc/ofconapp.cc.o
[ 28%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/ofstd/libsrc/ofcond.cc.o
[ 29%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/ofstd/libsrc/ofconfig.cc.o
[ 29%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/ofstd/libsrc/ofconsol.cc.o
[ 29%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/ofstd/libsrc/ofcrc32.cc.o
[ 30%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/ofstd/libsrc/ofdate.cc.o
[ 30%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/ofstd/libsrc/ofdatime.cc.o
[ 30%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/ofstd/libsrc/oferror.cc.o
[ 31%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/ofstd/libsrc/offile.cc.o
[ 31%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/ofstd/libsrc/offname.cc.o
[ 31%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/ofstd/libsrc/oflist.cc.o
[ 31%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/ofstd/libsrc/ofmath.cc.o
[ 32%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/ofstd/libsrc/ofstd.cc.o
[ 32%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/ofstd/libsrc/ofstring.cc.o
[ 32%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/ofstd/libsrc/oftempf.cc.o
[ 33%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/ofstd/libsrc/ofthread.cc.o
[ 33%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/ofstd/libsrc/oftime.cc.o
[ 33%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/ofstd/libsrc/oftimer.cc.o
[ 34%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/ofstd/libsrc/ofuuid.cc.o
[ 34%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/ofstd/libsrc/ofxml.cc.o
[ 34%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/apndimpl.cc.o
[ 34%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/appender.cc.o
[ 35%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/asyncap.cc.o
[ 35%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/clogger.cc.o
/wsi-to-dicom-converter/build/dcmtk-3.6.2/ofstd/libsrc/ofstd.cc: In static member function ‘static size_t OFStandard::searchDirectoryRecursively(const OFFilename&, OFList<OFFilename>&, const OFFilename&, const OFFilename&, OFBool)’:
/wsi-to-dicom-converter/build/dcmtk-3.6.2/ofstd/libsrc/ofstd.cc:1112:45: warning: ‘int readdir_r(DIR*, dirent*, dirent**)’ is deprecated [-Wdeprecated-declarations]
         while (!readdir_r(dirPtr, &d, &entry) && entry)
                                             ^
In file included from /wsi-to-dicom-converter/build/dcmtk-3.6.2/ofstd/libsrc/ofstd.cc:121:
/usr/include/dirent.h:183:12: note: declared here
 extern int readdir_r (DIR *__restrict __dirp,
            ^~~~~~~~~
/wsi-to-dicom-converter/build/dcmtk-3.6.2/ofstd/libsrc/ofstd.cc:1112:45: warning: ‘int readdir_r(DIR*, dirent*, dirent**)’ is deprecated [-Wdeprecated-declarations]
         while (!readdir_r(dirPtr, &d, &entry) && entry)
                                             ^
In file included from /wsi-to-dicom-converter/build/dcmtk-3.6.2/ofstd/libsrc/ofstd.cc:121:
/usr/include/dirent.h:183:12: note: declared here
 extern int readdir_r (DIR *__restrict __dirp,
            ^~~~~~~~~
[ 36%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/consap.cc.o
[ 36%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/config.cc.o
[ 36%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/cygwin32.cc.o
[ 36%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/env.cc.o
[ 36%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/factory.cc.o
[ 37%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/fileap.cc.o
[ 37%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/fileinfo.cc.o
[ 37%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/filter.cc.o
[ 38%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/globinit.cc.o
[ 38%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/hierarchy.cc.o
[ 38%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/hierlock.cc.o
[ 39%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/layout.cc.o
[ 39%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/lloguser.cc.o
[ 39%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/lockfile.cc.o
[ 39%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/log4judp.cc.o
[ 40%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/logevent.cc.o
[ 40%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/logger.cc.o
[ 40%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/logimpl.cc.o
[ 41%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/loglevel.cc.o
[ 41%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/loglog.cc.o
[ 41%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/logmacro.cc.o
[ 42%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/mdc.cc.o
[ 42%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/ndc.cc.o
[ 42%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/ntelogap.cc.o
[ 42%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/nullap.cc.o
[ 43%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/objreg.cc.o
[ 43%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/oflog.cc.o
[ 43%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/patlay.cc.o
[ 44%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/pointer.cc.o
[ 44%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/property.cc.o
[ 44%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/queue.cc.o
[ 45%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/rootlog.cc.o
[ 45%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/sleep.cc.o
[ 45%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/snprintf.cc.o
[ 45%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/sockbuff.cc.o
[ 46%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/socket.cc.o
[ 46%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/socketap.cc.o
[ 46%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/strccloc.cc.o
[ 47%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/strcloc.cc.o
[ 47%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/strhelp.cc.o
[ 47%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/striconv.cc.o
[ 48%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/syncprims.cc.o
[ 48%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/syslogap.cc.o
[ 48%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/threads.cc.o
[ 48%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/timehelp.cc.o
[ 49%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/tls.cc.o
[ 49%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/unixsock.cc.o
[ 49%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/version.cc.o
[ 50%] Building CXX object CMakeFiles/wsi2dcm.dir/dcmtk-3.6.2/oflog/libsrc/winconap.cc.o
[ 50%] Building CXX object CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/system/src/error_code.cpp.o
[ 50%] Building CXX object CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/atomic/src/lockpool.cpp.o
[ 50%] Building CXX object CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/thread/src/pthread/once.cpp.o
[ 51%] Building CXX object CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/thread/src/pthread/thread.cpp.o
[ 51%] Building CXX object CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/chrono/src/chrono.cpp.o
[ 51%] Building CXX object CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/program_options/src/value_semantic.cpp.o
[ 52%] Building CXX object CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/program_options/src/config_file.cpp.o
[ 52%] Building CXX object CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/program_options/src/convert.cpp.o
[ 52%] Building CXX object CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/program_options/src/options_description.cpp.o
[ 53%] Building CXX object CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/program_options/src/positional_options.cpp.o
[ 53%] Building CXX object CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/program_options/src/parsers.cpp.o
[ 53%] Building CXX object CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/program_options/src/variables_map.cpp.o
[ 53%] Building CXX object CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/program_options/src/cmdline.cpp.o
[ 54%] Building CXX object CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/program_options/src/utf8_codecvt_facet.cpp.o
[ 54%] Building CXX object CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/log/src/trivial.cpp.o
[ 54%] Building CXX object CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/log/src/attribute_name.cpp.o
[ 55%] Building CXX object CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/log/src/attribute_set.cpp.o
[ 55%] Building CXX object CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/log/src/attribute_value_set.cpp.o
[ 55%] Building CXX object CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/log/src/code_conversion.cpp.o
[ 56%] Building CXX object CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/log/src/date_time_format_parser.cpp.o
[ 56%] Building CXX object CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/log/src/default_attribute_names.cpp.o
[ 56%] Building CXX object CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/log/src/default_sink.cpp.o
[ 57%] Building CXX object CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/log/src/event.cpp.o
[ 57%] Building CXX object CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/log/src/dump.cpp.o
[ 57%] Building CXX object CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/log/src/exceptions.cpp.o
[ 57%] Building CXX object CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/log/src/format_parser.cpp.o
[ 58%] Building CXX object CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/log/src/global_logger_storage.cpp.o
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [CMakeFiles/wsi2dcm.dir/build.make:2260: CMakeFiles/wsi2dcm.dir/boost_1_69_0/libs/thread/src/pthread/thread.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:147: CMakeFiles/wsi2dcm.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
[==========] Running 24 tests from 10 test cases.
[----------] Global test environment set-up.
[----------] 4 tests from fileGeneration
[ RUN      ] fileGeneration.withoutConcatenation
[2021-06-04 19:19:09.900400] [0x00007fd63ff89a40] [error]   error: I/O suspension or premature end of stream
[       OK ] fileGeneration.withoutConcatenation (20 ms)
[ RUN      ] fileGeneration.withConcatenation
[       OK ] fileGeneration.withConcatenation (2 ms)
[ RUN      ] fileGeneration.fileSave
[       OK ] fileGeneration.fileSave (0 ms)
[ RUN      ] fileGeneration.fileSaveBatch
[       OK ] fileGeneration.fileSaveBatch (1 ms)
[----------] 4 tests from fileGeneration (23 ms total)

[----------] 4 tests from readJson
[ RUN      ] readJson.singleTag
[       OK ] readJson.singleTag (0 ms)
[ RUN      ] readJson.sequnceTag
[       OK ] readJson.sequnceTag (0 ms)
[ RUN      ] readJson.attributeTag
[       OK ] readJson.attributeTag (0 ms)
[ RUN      ] readJson.incorrectJson
[2021-06-04 19:19:09.904664] [0x00007fd63ff89a40] [warning] can't parse JSON
[2021-06-04 19:19:09.904701] [0x00007fd63ff89a40] [warning] * Line 1, Column 1
  Syntax error: value, object or array expected.

[       OK ] readJson.incorrectJson (0 ms)
[----------] 4 tests from readJson (0 ms total)

[----------] 1 test from insertBaseImageTagsTest
[ RUN      ] insertBaseImageTagsTest.correctInsert
[       OK ] insertBaseImageTagsTest.correctInsert (0 ms)
[----------] 1 test from insertBaseImageTagsTest (0 ms total)

[----------] 1 test from insertStaticTagsTest
[ RUN      ] insertStaticTagsTest.correctInsert
[       OK ] insertStaticTagsTest.correctInsert (0 ms)
[----------] 1 test from insertStaticTagsTest (0 ms total)

[----------] 1 test from insertIdsTest
[ RUN      ] insertIdsTest.correctInsert
[       OK ] insertIdsTest.correctInsert (0 ms)
[----------] 1 test from insertIdsTest (0 ms total)

[----------] 1 test from insertMultiFrameTagsTest
[ RUN      ] insertMultiFrameTagsTest.correctInsert
[       OK ] insertMultiFrameTagsTest.correctInsert (0 ms)
[----------] 1 test from insertMultiFrameTagsTest (0 ms total)

[----------] 3 tests from sliceFrame
[ RUN      ] sliceFrame.jpeg
[2021-06-04 19:19:09.928805] [0x00007fd63ff89a40] [debug]    frame size: 0kb
[       OK ] sliceFrame.jpeg (23 ms)
[ RUN      ] sliceFrame.jpeg2000Scaling
[2021-06-04 19:19:09.993874] [0x00007fd63ff89a40] [debug]    frame size: 11kb
[       OK ] sliceFrame.jpeg2000Scaling (66 ms)
[ RUN      ] sliceFrame.rawData
[2021-06-04 19:19:10.003770] [0x00007fd63ff89a40] [debug]    frame size: 29kb
[       OK ] sliceFrame.rawData (10 ms)
[----------] 3 tests from sliceFrame (99 ms total)

[----------] 2 tests from geometryTest
[ RUN      ] geometryTest.frameSizeWithoutRetile
[       OK ] geometryTest.frameSizeWithoutRetile (0 ms)
[ RUN      ] geometryTest.frameSizeWithRetile
[       OK ] geometryTest.frameSizeWithRetile (0 ms)
[----------] 2 tests from geometryTest (0 ms total)

[----------] 3 tests from readTiff
[ RUN      ] readTiff.simple
[2021-06-04 19:19:10.004536] [0x00007fd63ff89a40] [warning] threads parameter is less than 1, consuming all avalible threads
[2021-06-04 19:19:10.011803] [0x00007fd63ff89a40] [info]    dicomization is started
[2021-06-04 19:19:10.012215] [0x00007fd63ff89a40] [debug]   level size: 2220 2967 1
[2021-06-04 19:19:10.012229] [0x00007fd63ff89a40] [debug]   x: 0 y: 0 level: 0
[2021-06-04 19:19:10.012265] [0x00007fd63ff89a40] [debug]   x: 500 y: 0 level: 0
[2021-06-04 19:19:10.012298] [0x00007fd63ff89a40] [debug]   x: 1000 y: 0 level: 0
[2021-06-04 19:19:10.012336] [0x00007fd63ff89a40] [debug]   x: 1500 y: 0 level: 0
[2021-06-04 19:19:10.012370] [0x00007fd63ff89a40] [debug]   x: 2000 y: 0 level: 0
[2021-06-04 19:19:10.012404] [0x00007fd63ff89a40] [debug]   x: 0 y: 500 level: 0
[2021-06-04 19:19:10.012530] [0x00007fd63ff89a40] [debug]   x: 500 y: 500 level: 0
[2021-06-04 19:19:10.012550] [0x00007fd63ff89a40] [debug]   x: 1000 y: 500 level: 0
[2021-06-04 19:19:10.012569] [0x00007fd63ff89a40] [debug]   x: 1500 y: 500 level: 0
[2021-06-04 19:19:10.012605] [0x00007fd63ff89a40] [debug]   x: 2000 y: 500 level: 0
[2021-06-04 19:19:10.012646] [0x00007fd63ff89a40] [debug]   x: 0 y: 1000 level: 0
[2021-06-04 19:19:10.012691] [0x00007fd63ff89a40] [debug]   x: 500 y: 1000 level: 0
[2021-06-04 19:19:10.012737] [0x00007fd63ff89a40] [debug]   x: 1000 y: 1000 level: 0
[2021-06-04 19:19:10.012784] [0x00007fd63ff89a40] [debug]   x: 1500 y: 1000 level: 0
[2021-06-04 19:19:10.012824] [0x00007fd63ff89a40] [debug]   x: 2000 y: 1000 level: 0
[2021-06-04 19:19:10.012860] [0x00007fd63ff89a40] [debug]   x: 0 y: 1500 level: 0
[2021-06-04 19:19:10.012904] [0x00007fd63ff89a40] [debug]   x: 500 y: 1500 level: 0
[2021-06-04 19:19:10.012956] [0x00007fd63ff89a40] [debug]   x: 1000 y: 1500 level: 0
[2021-06-04 19:19:10.013001] [0x00007fd63ff89a40] [debug]   x: 1500 y: 1500 level: 0
[2021-06-04 19:19:10.013048] [0x00007fd63ff89a40] [debug]   x: 2000 y: 1500 level: 0
[2021-06-04 19:19:10.013090] [0x00007fd63ff89a40] [debug]   x: 0 y: 2000 level: 0
[2021-06-04 19:19:10.013127] [0x00007fd63ff89a40] [debug]   x: 500 y: 2000 level: 0
[2021-06-04 19:19:10.013158] [0x00007fd63ff89a40] [debug]   x: 1000 y: 2000 level: 0
[2021-06-04 19:19:10.013171] [0x00007fd63ff89a40] [debug]   x: 1500 y: 2000 level: 0
[2021-06-04 19:19:10.013183] [0x00007fd63ff89a40] [debug]   x: 2000 y: 2000 level: 0
[2021-06-04 19:19:10.013189] [0x00007fd63ff89a40] [debug]   x: 0 y: 2500 level: 0
[2021-06-04 19:19:10.013241] [0x00007fd63ff89a40] [debug]   x: 500 y: 2500 level: 0
[2021-06-04 19:19:10.013273] [0x00007fd63ff89a40] [debug]   x: 1000 y: 2500 level: 0
[2021-06-04 19:19:10.013285] [0x00007fd63ff89a40] [debug]   x: 1500 y: 2500 level: 0
[2021-06-04 19:19:10.013299] [0x00007fd63ff89a40] [debug]   x: 2000 y: 2500 level: 0
[2021-06-04 19:19:10.069036] [0x00007fd63c9fd700] [debug]    frame size: 7kb
[2021-06-04 19:19:10.071195] [0x00007fd63e200700] [debug]    frame size: 40kb
[2021-06-04 19:19:10.072033] [0x00007fd63ea01700] [debug]    frame size: 21kb
[2021-06-04 19:19:10.073732] [0x00007fd63d1fe700] [debug]    frame size: 6kb
[2021-06-04 19:19:10.075525] [0x00007fd63f202700] [debug]    frame size: 7kb
[2021-06-04 19:19:10.075687] [0x00007fd63d9ff700] [debug]    frame size: 6kb
[2021-06-04 19:19:10.086691] [0x00007fd63d1fe700] [debug]    frame size: 7kb
[2021-06-04 19:19:10.089635] [0x00007fd63ea01700] [debug]    frame size: 24kb
[2021-06-04 19:19:10.093083] [0x00007fd63c9fd700] [debug]    frame size: 20kb
[2021-06-04 19:19:10.094327] [0x00007fd63e200700] [debug]    frame size: 63kb
[2021-06-04 19:19:10.094562] [0x00007fd63f202700] [debug]    frame size: 14kb
[2021-06-04 19:19:10.095168] [0x00007fd63d9ff700] [debug]    frame size: 38kb
[2021-06-04 19:19:10.104916] [0x00007fd63c9fd700] [debug]    frame size: 6kb
[2021-06-04 19:19:10.106327] [0x00007fd63d1fe700] [debug]    frame size: 69kb
[2021-06-04 19:19:10.107161] [0x00007fd63ea01700] [debug]    frame size: 51kb
[2021-06-04 19:19:10.108264] [0x00007fd63e200700] [debug]    frame size: 6kb
[2021-06-04 19:19:10.113444] [0x00007fd63f202700] [debug]    frame size: 43kb
[2021-06-04 19:19:10.116033] [0x00007fd63d9ff700] [debug]    frame size: 74kb
[2021-06-04 19:19:10.117155] [0x00007fd63d1fe700] [debug]    frame size: 6kb
[2021-06-04 19:19:10.124139] [0x00007fd63ea01700] [debug]    frame size: 5kb
[2021-06-04 19:19:10.127090] [0x00007fd63d1fe700] [debug]    frame size: 6kb
[2021-06-04 19:19:10.127522] [0x00007fd63e200700] [debug]    frame size: 56kb
[2021-06-04 19:19:10.127827] [0x00007fd63c9fd700] [debug]    frame size: 37kb
[2021-06-04 19:19:10.133649] [0x00007fd63d9ff700] [debug]    frame size: 63kb
[2021-06-04 19:19:10.134573] [0x00007fd63f202700] [debug]    frame size: 71kb
[2021-06-04 19:19:10.139342] [0x00007fd63ea01700] [debug]    frame size: 6kb
[2021-06-04 19:19:10.143490] [0x00007fd63d9ff700] [debug]    frame size: 6kb
[2021-06-04 19:19:10.144051] [0x00007fd63c9fd700] [debug]    frame size: 40kb
[2021-06-04 19:19:10.146331] [0x00007fd63d1fe700] [debug]    frame size: 79kb
[2021-06-04 19:19:10.148776] [0x00007fd63e200700] [debug]    frame size: 80kb
[2021-06-04 19:19:10.239733] [0x00007fd63ff89a40] [info]    dicomization is done
[       OK ] readTiff.simple (236 ms)
[ RUN      ] readTiff.withJson
[2021-06-04 19:19:10.240141] [0x00007fd63ff89a40] [warning] threads parameter is less than 1, consuming all avalible threads
[2021-06-04 19:19:10.251194] [0x00007fd63ff89a40] [info]    dicomization is started
[2021-06-04 19:19:10.251835] [0x00007fd63ff89a40] [debug]   level size: 2220 2967 1
[2021-06-04 19:19:10.251864] [0x00007fd63ff89a40] [debug]   x: 0 y: 0 level: 0
[2021-06-04 19:19:10.592897] [0x00007fd63f202700] [debug]    frame size: 19297kb
[2021-06-04 19:19:10.722413] [0x00007fd63ff89a40] [info]    dicomization is done
[       OK ] readTiff.withJson (482 ms)
[ RUN      ] readTiff.multiFile
[2021-06-04 19:19:10.722844] [0x00007fd63ff89a40] [warning] threads parameter is less than 1, consuming all avalible threads
[2021-06-04 19:19:10.733031] [0x00007fd63ff89a40] [info]    dicomization is started
[2021-06-04 19:19:10.733987] [0x00007fd63ff89a40] [debug]   level size: 2220 2967 1
[2021-06-04 19:19:10.734048] [0x00007fd63ff89a40] [debug]   x: 0 y: 0 level: 0
[2021-06-04 19:19:10.734098] [0x00007fd63ff89a40] [debug]   x: 500 y: 0 level: 0
[2021-06-04 19:19:10.734166] [0x00007fd63ff89a40] [debug]   x: 1000 y: 0 level: 0
[2021-06-04 19:19:10.734248] [0x00007fd63ff89a40] [debug]   x: 1500 y: 0 level: 0
[2021-06-04 19:19:10.734296] [0x00007fd63ff89a40] [debug]   x: 2000 y: 0 level: 0
[2021-06-04 19:19:10.734348] [0x00007fd63ff89a40] [debug]   x: 0 y: 500 level: 0
[2021-06-04 19:19:10.734374] [0x00007fd63ff89a40] [debug]   x: 500 y: 500 level: 0
[2021-06-04 19:19:10.734399] [0x00007fd63ff89a40] [debug]   x: 1000 y: 500 level: 0
[2021-06-04 19:19:10.734684] [0x00007fd63ff89a40] [debug]   x: 1500 y: 500 level: 0
[2021-06-04 19:19:10.734788] [0x00007fd63ff89a40] [debug]   x: 2000 y: 500 level: 0
[2021-06-04 19:19:10.734822] [0x00007fd63ff89a40] [debug]   x: 0 y: 1000 level: 0
[2021-06-04 19:19:10.734907] [0x00007fd63ff89a40] [debug]   x: 500 y: 1000 level: 0
[2021-06-04 19:19:10.734967] [0x00007fd63ff89a40] [debug]   x: 1000 y: 1000 level: 0
[2021-06-04 19:19:10.734996] [0x00007fd63ff89a40] [debug]   x: 1500 y: 1000 level: 0
[2021-06-04 19:19:10.735071] [0x00007fd63ff89a40] [debug]   x: 2000 y: 1000 level: 0
[2021-06-04 19:19:10.735104] [0x00007fd63ff89a40] [debug]   x: 0 y: 1500 level: 0
[2021-06-04 19:19:10.735132] [0x00007fd63ff89a40] [debug]   x: 500 y: 1500 level: 0
[2021-06-04 19:19:10.735141] [0x00007fd63ff89a40] [debug]   x: 1000 y: 1500 level: 0
[2021-06-04 19:19:10.735173] [0x00007fd63ff89a40] [debug]   x: 1500 y: 1500 level: 0
[2021-06-04 19:19:10.735203] [0x00007fd63ff89a40] [debug]   x: 2000 y: 1500 level: 0
[2021-06-04 19:19:10.735285] [0x00007fd63ff89a40] [debug]   x: 0 y: 2000 level: 0
[2021-06-04 19:19:10.735332] [0x00007fd63ff89a40] [debug]   x: 500 y: 2000 level: 0
[2021-06-04 19:19:10.735382] [0x00007fd63ff89a40] [debug]   x: 1000 y: 2000 level: 0
[2021-06-04 19:19:10.735451] [0x00007fd63ff89a40] [debug]   x: 1500 y: 2000 level: 0
[2021-06-04 19:19:10.735574] [0x00007fd63ff89a40] [debug]   x: 2000 y: 2000 level: 0
[2021-06-04 19:19:10.735724] [0x00007fd63ff89a40] [debug]   x: 0 y: 2500 level: 0
[2021-06-04 19:19:10.735783] [0x00007fd63ff89a40] [debug]   x: 500 y: 2500 level: 0
[2021-06-04 19:19:10.735913] [0x00007fd63ff89a40] [debug]   x: 1000 y: 2500 level: 0
[2021-06-04 19:19:10.735976] [0x00007fd63ff89a40] [debug]   x: 1500 y: 2500 level: 0
[2021-06-04 19:19:10.736214] [0x00007fd63ff89a40] [debug]   x: 2000 y: 2500 level: 0
[2021-06-04 19:19:10.772712] [0x00007fd63ea01700] [debug]    frame size: 17kb
[2021-06-04 19:19:10.787805] [0x00007fd63d1fe700] [debug]    frame size: 37kb
[2021-06-04 19:19:10.791001] [0x00007fd63c9fd700] [debug]    frame size: 54kb
[2021-06-04 19:19:10.794990] [0x00007fd63d9ff700] [debug]    frame size: 40kb
[2021-06-04 19:19:10.816239] [0x00007fd63f202700] [debug]    frame size: 126kb
[2021-06-04 19:19:10.825914] [0x00007fd63d9ff700] [debug]    frame size: 28kb
[2021-06-04 19:19:10.834427] [0x00007fd63ea01700] [debug]    frame size: 111kb
[2021-06-04 19:19:10.844990] [0x00007fd63e200700] [debug]    frame size: 278kb
[2021-06-04 19:19:10.853731] [0x00007fd63c9fd700] [debug]    frame size: 149kb
[2021-06-04 19:19:10.883400] [0x00007fd63d9ff700] [debug]    frame size: 96kb
[2021-06-04 19:19:10.913714] [0x00007fd63d9ff700] [debug]    frame size: 20kb
[2021-06-04 19:19:10.915819] [0x00007fd63ea01700] [debug]    frame size: 223kb
[2021-06-04 19:19:10.921069] [0x00007fd63d1fe700] [debug]    frame size: 417kb
[2021-06-04 19:19:10.947148] [0x00007fd63d9ff700] [debug]    frame size: 20kb
[2021-06-04 19:19:10.959878] [0x00007fd63c9fd700] [debug]    frame size: 323kb
[2021-06-04 19:19:10.982734] [0x00007fd63e200700] [debug]    frame size: 492kb
[2021-06-04 19:19:10.984348] [0x00007fd63c9fd700] [debug]    frame size: 20kb
[2021-06-04 19:19:10.993513] [0x00007fd63ea01700] [debug]    frame size: 233kb
[2021-06-04 19:19:11.012270] [0x00007fd63c9fd700] [debug]    frame size: 20kb
[2021-06-04 19:19:11.031411] [0x00007fd63d9ff700] [debug]    frame size: 225kb
[2021-06-04 19:19:11.057938] [0x00007fd63d1fe700] [debug]    frame size: 479kb
[2021-06-04 19:19:11.064672] [0x00007fd63d9ff700] [debug]    frame size: 19kb
[2021-06-04 19:19:11.090433] [0x00007fd63d1fe700] [debug]    frame size: 14kb
[2021-06-04 19:19:11.107684] [0x00007fd63ea01700] [debug]    frame size: 327kb
[2021-06-04 19:19:11.133305] [0x00007fd63ea01700] [debug]    frame size: 12kb
[2021-06-04 19:19:11.140556] [0x00007fd63f202700] [debug]    frame size: 381kb
[2021-06-04 19:19:11.163200] [0x00007fd63c9fd700] [debug]    frame size: 496kb
[2021-06-04 19:19:11.175738] [0x00007fd63d1fe700] [debug]    frame size: 237kb
[2021-06-04 19:19:11.214878] [0x00007fd63d9ff700] [debug]    frame size: 465kb
[2021-06-04 19:19:11.219694] [0x00007fd63e200700] [debug]    frame size: 486kb
[2021-06-04 19:19:11.243785] [0x00007fd63ff89a40] [info]    dicomization is done
[       OK ] readTiff.multiFile (521 ms)
[----------] 3 tests from readTiff (1239 ms total)

[----------] 4 tests from compressionString
[ RUN      ] compressionString.jpeg
[       OK ] compressionString.jpeg (0 ms)
[ RUN      ] compressionString.jpeg2000
[       OK ] compressionString.jpeg2000 (0 ms)
[ RUN      ] compressionString.none
[       OK ] compressionString.none (0 ms)
[ RUN      ] compressionString.unknown
[       OK ] compressionString.unknown (0 ms)
[----------] 4 tests from compressionString (0 ms total)

[----------] Global test environment tear-down
[==========] 24 tests from 10 test cases ran. (1363 ms total)
[  PASSED  ] 24 tests.
Reading package lists... Done
Building dependency tree
Reading state information... Done
dcmtk is already the newest version (3.6.4-2.1).
wget is already the newest version (1.20.1-1.1).
0 upgraded, 0 newly installed, 0 to remove and 6 not upgraded.
W: Target Packages (universe/binary-amd64/Packages) is configured multiple times in /etc/apt/sources.list:7 and /etc/apt/sources.list:8
W: Target Packages (universe/binary-all/Packages) is configured multiple times in /etc/apt/sources.list:7 and /etc/apt/sources.list:8
[2021-06-04 19:19:12.441367] [0x00007f2e5af4aa40] [warning] studyId is going to be generated
[2021-06-04 19:19:12.441551] [0x00007f2e5af4aa40] [warning] seriesId is going to be generated
[2021-06-04 19:19:12.441574] [0x00007f2e5af4aa40] [warning] threads parameter is less than 1, consuming all avalible threads
[2021-06-04 19:19:12.441624] [0x00007f2e5af4aa40] [warning] batch parameter is not set, batch is unlimited
[2021-06-04 19:19:12.452033] [0x00007f2e5af4aa40] [info]    dicomization is started
[2021-06-04 19:19:12.864890] [0x00007f2e5af4aa40] [info]    dicomization is done
[2021-06-04 19:19:12.919822] [0x00007ff63484ba40] [warning] studyId is going to be generated
[2021-06-04 19:19:12.919957] [0x00007ff63484ba40] [warning] seriesId is going to be generated
[2021-06-04 19:19:12.919984] [0x00007ff63484ba40] [warning] threads parameter is less than 1, consuming all avalible threads
[2021-06-04 19:19:12.920023] [0x00007ff63484ba40] [warning] batch parameter is not set, batch is unlimited
[2021-06-04 19:19:12.930921] [0x00007ff63484ba40] [info]    dicomization is started
[2021-06-04 19:19:13.174029] [0x00007ff63484ba40] [info]    dicomization is done
[2021-06-04 19:19:13.223914] [0x00007f9a1e1f0a40] [warning] studyId is going to be generated
[2021-06-04 19:19:13.223995] [0x00007f9a1e1f0a40] [warning] seriesId is going to be generated
[2021-06-04 19:19:13.224016] [0x00007f9a1e1f0a40] [warning] threads parameter is less than 1, consuming all avalible threads
[2021-06-04 19:19:13.224037] [0x00007f9a1e1f0a40] [warning] batch parameter is not set, batch is unlimited
[2021-06-04 19:19:13.235550] [0x00007f9a1e1f0a40] [info]    dicomization is started
[2021-06-04 19:19:13.551304] [0x00007f9a1e1f0a40] [info]    dicomization is done
[2021-06-04 19:19:13.600793] [0x00007fbfd0d46a40] [warning] studyId is going to be generated
[2021-06-04 19:19:13.600868] [0x00007fbfd0d46a40] [warning] seriesId is going to be generated
[2021-06-04 19:19:13.600884] [0x00007fbfd0d46a40] [warning] threads parameter is less than 1, consuming all avalible threads
[2021-06-04 19:19:13.600900] [0x00007fbfd0d46a40] [warning] batch parameter is not set, batch is unlimited
[2021-06-04 19:19:13.611818] [0x00007fbfd0d46a40] [info]    dicomization is started
[2021-06-04 19:19:14.026135] [0x00007fbfd0d46a40] [info]    dicomization is done

danielbeaudreau commented on src/geometryUtils.cpp:
```diff
@@ -11,46 +10,46 @@
-      adjustFrameToLevel(frameHeightDownsampled, frameWidhtDownsampled,
-                         levelHeight);
-    }
+  if (levelWidth < *frameWidthDownsampled) {
+    /*
+        level_frameWidth could optimally be set to *levelWidthDownsampled
+        However, a bug in JPEG2000 Codec segfaults if the size of the allocated
+       frame being compressed is too small.  JPEG Codec is fine.  The max is a
+       bandaid fix for the isseue. JPEG2000 codec should be fixed and the max
+       can be removed here and in levelHeight < To reproduce bug: remove max,
+       run JPEG2000 test case in end-to-end tests.
+    */
+    *frameWidthDownsampled = levelWidth;
+    *level_frameWidth = std::max<int64_t>(*levelWidthDownsampled, 50);
+  }
+  if (levelHeight < *frameHeightDownsampled) {
+    *frameHeightDownsampled = levelHeight;
+    *level_frameHeight = std::max<int64_t>(*levelHeightDownsampled, 50);
```
Not very familiar with this code, could you add a bit more explanation on these changes and what the bug you're fixing is? Why do frame width and height need to be updated here? From what I understand these two parameters are used to set the final dicom frame width and height by rescaling the boost view in frame.cpp

danielbeaudreau commented on src/geometryUtils.h:
```diff
@@ -15,29 +15,25 @@
+// frameWidthDownsampled, frameHeightDownsampled - size of frame to
 //                                                 get from source level
-// levelWidhtDownsampled, levelHeightDownsampled - size of level
+// levelWidthDownsampled, levelHeightDownsampled - size of level
 //                                                 according to downsampling
```
nit: please add a comment for the new parameters

danielbeaudreau commented on src/wsiToDcm.cpp:
```diff
@@ -196,18 +198,33 @@ int WsiToDcm::dicomizeTiff(std::string inputFile, std::string outputFileMask,
         assert(osr != nullptr && openslide_get_error(osr) == nullptr);
-        BOOST_LOG_TRIVIAL(debug)
-            << "x: " << x << " y: " << y << " level: " << level;
+        // BOOST_LOG_TRIVIAL(debug)
+        //    << "x: " << x << " y: " << y << " level: " << level;
```
nit: Instead of commenting out this line it can be removed if it's not useful.

Philbrick replied:
Dan I found it..

danielbeaudreau commented:
/gcbrun

danielbeaudreau commented:
The presubmit checks that the end to end .sh script returns 0. Could you verify that when you run the script locally it returns 0? (you can check with `echo $?` after running the command)

Philbrick committed:
Fix's resizing issues for downsampleing to images < 1 frame in size

Fixes issue downsizing images to sizs less than default frame size.  Old code incorrectly scaled frames resulting in imaging being incorrectly downsampled.  Imaging was clipped with reference to the origional frame.

 Updates test cases to reflect changes.  Provides a workaround for issue saving very small images in JPEG2000 codec.   For these images saves  as RAW.

danielbeaudreau commented:
/gcbrun

danielbeaudreau commented:
please fix linter errors

././src/geometryUtils.cpp:25:  Is this a non-const reference? If so, make const or use a pointer: DCM_Compression &compression  [runtime/references] [2]
././src/geometryUtils.cpp:38:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
././src/geometryUtils.h:32:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
././src/geometryUtils.h:34:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
././src/geometryUtils.h:35:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
././src/geometryUtils.h:43:  Is this a non-const reference? If so, make const or use a pointer: DCM_Compression &compression  [runtime/references] [2]
././src/wsiToDcm.cpp:18:  Found C system header after C++ system header. Should be: wsiToDcm.h, c system, c++ system, other.  [build/include_order] [4]
././src/wsiToDcm.cpp:19:  Found C system header after C++ system header. Should be: wsiToDcm.h, c system, c++ system, other.  [build/include_order] [4]
././src/wsiToDcm.cpp:20:  Found C system header after C++ system header. Should be: wsiToDcm.h, c system, c++ system, other.  [build/include_order] [4]
././src/wsiToDcm.cpp:21:  Found C system header after C++ system header. Should be: wsiToDcm.h, c system, c++ system, other.  [build/include_order] [4]
././src/wsiToDcm.cpp:22:  Found C system header after C++ system header. Should be: wsiToDcm.h, c system, c++ system, other.  [build/include_order] [4]
././src/wsiToDcm.cpp:23:  Found C system header after C++ system header. Should be: wsiToDcm.h, c system, c++ system, other.  [build/include_order] [4]
././src/wsiToDcm.cpp:24:  Found C system header after C++ system header. Should be: wsiToDcm.h, c system, c++ system, other.  [build/include_order] [4]

Philbrick committed:
Fix C++ linting errors

danielbeaudreau commented:
/gcbrun

danielbeaudreau commented:
more errors: 

././src/main.cpp:18:  Found C system header after C++ system header. Should be: main.h, c system, c++ system, other.  [build/include_order] [4]
././src/wsiToDcm.cpp:26:  Found C system header after C++ system header. Should be: wsiToDcm.h, c system, c++ system, other.  [build/include_order] [4]
././src/wsiToDcm.cpp:27:  Found C system header after C++ system header. Should be: wsiToDcm.h, c system, c++ system, other.  [build/include_order] [4]
././src/wsiToDcm.cpp:28:  Found C system header after C++ system header. Should be: wsiToDcm.h, c system, c++ system, other.  [build/include_order] [4]
././src/wsiToDcm.cpp:29:  Found C system header after C++ system header. Should be: wsiToDcm.h, c system, c++ system, other.  [build/include_order] [4]
././src/wsiToDcm.cpp:30:  Found C system header after C++ system header. Should be: wsiToDcm.h, c system, c++ system, other.  [build/include_order] [4]
././src/wsiToDcm.cpp:31:  Found C system header after C++ system header. Should be: wsiToDcm.h, c system, c++ system, other.  [build/include_order] [4]
././src/wsiToDcm.cpp:32:  Found C system header after C++ system header. Should be: wsiToDcm.h, c system, c++ system, other.  [build/include_order] [4]
././src/wsiToDcm.h:21:  Found C system header after C++ system header. Should be: wsiToDcm.h, c system, c++ system, other.  [build/include_order] [4]

Philbrick committed:
fix lint include order

fix include order

danielbeaudreau commented:
/gcbrun
